apiserver/common/machine_test.go

```diff
@@ -239,10 +240,53 @@ func (s *machineSuite) TestMachineInstanceInfoWithSetDisplayName(c *gc.C) {
 	})
 }
 
+func (s *machineSuite) TestMachineInstanceInfoWithHAPrimary(c *gc.C) {
+	st := mockState{
+		machines: map[string]*mockMachine{
+			"1": {
+				id:          "1",
+				instId:      "123",
+				displayName: "snowflake",
+				status:      status.Down,
+			},
+		},
+		controllerNodes: map[string]*mockControllerNode{
+			"1": {
+				id:        "1",
+				hasVote:   true,
+				wantsVote: true,
+			},
+			"2": {
+				id:        "1",
+				hasVote:   true,
+				wantsVote: true,
+			},
+		},
+		haPimaryMachineF: func() (names.MachineTag, error) {
+			return names.NewMachineTag("1"), nil
+		},
+	}
+	info, err := common.ModelMachineInfo(&st)
+	c.Assert(err, jc.ErrorIsNil)
+	_true := true
+	c.Assert(info, jc.DeepEquals, []params.ModelMachineInfo{
+		{
+			Id:          "1",
+			InstanceId:  "123",
+			DisplayName: "snowflake",
+			Status:      "down",
+			HasVote:     true,
+			WantsVote:   true,
+			HAPrimary:   &_true,
+		},
+	})
+}
+
 type mockState struct {
 	common.ModelManagerBackend
-	machines        map[string]*mockMachine
-	controllerNodes map[string]*mockControllerNode
+	machines         map[string]*mockMachine
+	controllerNodes  map[string]*mockControllerNode
+	haPimaryMachineF func() (names.MachineTag, error)
```

Comment:
typo: haPimaryMachineF

Reply:
Oooh ! eye of an eagle :) Thank you for spotting.